torch/csrc/lazy/backend/backend_interface.h

```diff
@@ -40,6 +41,8 @@ class TORCH_API BackendImplInterface {
 
   virtual const IrBuilder* GetIrBuilder() const = 0;
 
+  virtual bool ShouldSyncTensor(const LazyTensorPtr tensor) const;
```

Comment:
I can't really see a functionality issue with the change you propose, as the default implementation is the same and just uses one more indirection/functioncall so probably no perf impact.

But I feel like it's not too obvious how this API is supposed to be used or what the contract between frontend/backend is.  What does it mean if a backend answers 'no' on a tensor?  What if it answers no for every tensor?  Will this impact what the user can do with the frontend?  We should make this clearer.

Reply:
> What does it mean if a backend answers 'no' on a tensor?

Effectively it means that the end user doesn't have access to the the value of that tensor even after mark step. This is unfortunately a "property" of our hardware, so its just easier to prune those outputs out of the graph rather than have to deal with it further down in the compile/execute pipeline.

>  What if it answers no for every tensor?

If it answers no for every tensor, then there effectively is no traced graph and thus nothing to run. As such, this is a case that wouldn't practically come up.

> Will this impact what the user can do with the frontend?

Of course. It is up to the vendor to make it known to their users what limitations their hardware has.